test/access/manager/AccessManaged.test.js

```diff
@@ -0,0 +1,133 @@
+const { expectEvent, time, expectRevert } = require('@openzeppelin/test-helpers');
+const { selector } = require('../../helpers/methods');
+const { expectRevertCustomError } = require('../../helpers/customError');
+const {
+  time: { setNextBlockTimestamp },
+} = require('@nomicfoundation/hardhat-network-helpers');
+const { impersonate } = require('../../helpers/account');
+const { CONSUMING_SCHEDULE_STORAGE_SLOT } = require('../../helpers/access-manager');
+
+const AccessManaged = artifacts.require('$AccessManagedTarget');
+const AccessManager = artifacts.require('$AccessManager');
+
+contract('AccessManaged', function (accounts) {
+  const [admin, roleMember, other] = accounts;
+
+  beforeEach(async function () {
+    this.authority = await AccessManager.new(admin);
+    this.managed = await AccessManaged.new(this.authority.address);
+  });
+
+  it('sets authority and emits AuthorityUpdated event during construction', async function () {
+    await expectEvent.inConstruction(this.managed, 'AuthorityUpdated', {
+      authority: this.authority.address,
+    });
+    expect(await this.managed.authority()).to.eq(this.authority.address);
+  });
+
+  describe('restricted modifier', function () {
+    const method = 'fnRestricted()';
+
+    beforeEach(async function () {
+      this.selector = selector(method);
+      this.role = web3.utils.toBN(42);
+      await this.authority.$_setTargetFunctionRole(this.managed.address, this.selector, this.role);
+      await this.authority.$_grantRole(this.role, roleMember, 0, 0);
+    });
+
+    it('succeeds when role is granted without execution delay', async function () {
+      await this.managed.methods[method]({ from: roleMember });
+    });
+
+    it('reverts when role is not granted', async function () {
+      await expectRevertCustomError(this.managed.methods[method]({ from: other }), 'AccessManagedUnauthorized', [
+        other,
+      ]);
+    });
+
+    it('panics in short calldata', async function () {
+      // We avoid adding the `restricted` modifier to the fallback function because other tests may depend on it
+      // being accessible without restrictions. We check for the internal `_checkCanCall` instead.
+      await expectRevert.unspecified(this.managed.$_checkCanCall(other, '0x1234'));
+    });
+
+    describe('when role is granted with execution delay', function () {
+      beforeEach(async function () {
+        const executionDelay = web3.utils.toBN(911);
+        await this.authority.$_grantRole(this.role, roleMember, 0, executionDelay);
+      });
+
+      it('reverts if the operation is not scheduled', async function () {
+        const calldata = await this.managed.contract.methods[method]().encodeABI();
```

Comment:
This should not require an await, does it ?
```suggestion
        const calldata = this.managed.contract.methods[method]().encodeABI();
```

Reply:
there are other calls to `encodeABI()` that this may apply to.